tornado/web.py

```diff
@@ -2278,7 +2278,11 @@ def initialize(self, url, permanent=True):
         self._permanent = permanent
 
     def get(self, *args):
-        self.redirect(self._url.format(*args), permanent=self._permanent)
+        to_url = self._url.format(*args)
+        if self.request.query:
+            to_url = httputil.url_concat(
+                to_url, urlparse.parse_qsl(self.request.query))
```

Comment:
I prefered the the way like `urllib.parse.parse_qsl(urllib.parse.urlencode(qs, doseq=True))`, but the generator function is more straightforward and clear.